### DISCUSSION TITLE
Consider replacing lambda conversion from `Expression` to `LambdaExpression`

### DISCUSSION THREAD

**Discussion by mburbea** [dotnet/csharplang#5191] [Language Ideas]
Consider the following
```cs
static void M(LambdaExpression l){}
static void C() => M(() => 7);
```
This gets the error message:
> CS1660	Cannot convert lambda expression to type 'LambdaExpression' because it is not a delegate type

The error message is of course confusing as if I add this overload then it will compile.

```cs
static void M(Expression e){}
```

But this seems a bit odd to me, as `Expression` is not necessarily an expression tree that represents an invokable function. I think this change can be made before .net 6 is out. There should support for `LambdaExpression` as there is support for `Expression<TDelegate>`, or my non-generic overload needs to have the following line of code.

```cs
static void M(Expression e)
{
    if(e is null) throw new ArgumentNullException(...);
    if(e is not LambdaExpression) throw new ArgumentExceptionException(...);
     //logic here that relies on it being an invokable expression
}
```

**Comment by cston** [comment]
@mburbea, thanks for reporting this issue.

This has been fixed in https://github.com/dotnet/roslyn/pull/56200. Lambda expressions are now implicitly convertible to `LambdaExpression`.
